testing/tests/ui/default.stderr

```diff
@@ -0,0 +1,88 @@
+error: the `default` filter requires a variable name on its left-hand side
+ --> DefaultLhsExpr.html:1:11
+       "| default(2) }}"
+ --> tests/ui/default.rs:6:35
+  |
+6 | #[template(ext = "html", source = "{{ (1 + 1) | default(2) }}")]
+  |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: `default_value` argument is missing when calling `default` filter; its arguments are: (default_value, [boolean])
+ --> DefaultNoDefaultValue.html:1:7
+       "| default }}"
+  --> tests/ui/default.rs:10:35
+   |
+10 | #[template(ext = "html", source = "{{ var | default }}")]
+   |                                   ^^^^^^^^^^^^^^^^^^^^^
+
+error: `default_value` argument is missing when calling `default` filter; its arguments are: (default_value, [boolean])
+ --> DefaultNoDefaultValue2.html:1:7
+       "| default(boolean = true) }}"
+  --> tests/ui/default.rs:14:35
+   |
+14 | #[template(ext = "html", source = "{{ var | default(boolean = true) }}")]
+   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `default` filter takes a boolean literal as its optional second argument
+ --> DefaultNotABool.html:1:20
+       "2) }}"
+  --> tests/ui/default.rs:24:35
+   |
+24 | #[template(ext = "html", source = "{{ var | default(1, 2) }}")]
+   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `default` filter takes a boolean literal as its optional second argument
+ --> DefaultNotABool2.html:1:30
+       "2) }}"
+  --> tests/ui/default.rs:28:35
+   |
+28 | #[template(ext = "html", source = "{{ var | default(1, boolean = 2) }}")]
+   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: `fallback` argument is missing when calling `assigned_or` filter; its arguments are: (fallback)
+ --> AssignedOrNoDefaultValue.html:1:7
+       "| assigned_or }}"
+  --> tests/ui/default.rs:32:35
+   |
+32 | #[template(ext = "html", source = "{{ var | assigned_or }}")]
+   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `defined_or` filter requires a variable name on its left-hand side
+ --> DefinedOrLhsExpr.html:1:11
+       "| defined_or(2) }}"
+  --> tests/ui/default.rs:36:35
+   |
+36 | #[template(ext = "html", source = "{{ (1 + 1) | defined_or(2) }}")]
+   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: `fallback` argument is missing when calling `defined_or` filter; its arguments are: (fallback)
+ --> DefinedOrNoDefaultValue.html:1:7
+       "| defined_or }}"
+  --> tests/ui/default.rs:40:35
+   |
+40 | #[template(ext = "html", source = "{{ var | defined_or }}")]
+   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^
+
+error[E0277]: `Mutex<u32>` is not `|assigned_or` filterable
```

Comment:
I didn't know we could do that in rust, it looks super nice! (and will look even better once I add the spans to the generated `TokenStream` :3)

Reply:
The feature is already stable for over a year: <https://releases.rs/docs/1.78.0/>, but I am not sure if I ever saw it in the wild. I found it [in the standard library](https://doc.rust-lang.org/1.88.0/src/core/future/future.rs.html#32) and simply tested if library developers can use it, too. :)

Reply:
A lot of features are, but if I don't use them, I simply forget about them. ^^'